Mathlib/Geometry/Manifold/MFDeriv/SpecificFunctions.lean

```diff
@@ -660,6 +660,76 @@ theorem mfderiv_prod_eq_add_apply {f : M × M' → M''} {p : M × M'} {v : Tange
 
 end Prod
 
+section disjointUnion
+
+variable {M' : Type*} [TopologicalSpace M'] [ChartedSpace H M'] {n : WithTop ℕ∞}
+  {E' : Type*} [NormedAddCommGroup E'] [NormedSpace 𝕜 E'] {H' : Type*} [TopologicalSpace H']
+  {J : Type*} {J : ModelWithCorners 𝕜 E' H'}
+  {N N' : Type*} [TopologicalSpace N] [TopologicalSpace N'] [ChartedSpace H' N] [ChartedSpace H' N']
+
+open Topology
+
+variable {s : Set (M ⊕ M')} {p : M ⊕ M'}
+
+/-- In extended charts at `p`, `Sum.swap` looks like the identity near `p`. -/
+lemma writtenInExtChartAt_sumSwap_eventuallyEq_id :
+    writtenInExtChartAt I I p Sum.swap =ᶠ[𝓝[(extChartAt I p).symm ⁻¹' s ∩ range I]
+      extChartAt I p p] id := by
```

Comment:
I'm having trouble understanding why `s` is included here. I would have expected to express this as something like:
```lean
example :
    writtenInExtChartAt I I p Sum.swap =ᶠ[𝓝[range I] (I <| chartAt H p p)] id := by
  sorry
```
or even (via `eventually_nhdsWithin_iff`):
```lean
example :
    ∀ᶠ x in 𝓝 (I <| chartAt H p p), x ∈ range I → writtenInExtChartAt I I p Sum.swap x = x := by
  sorry
```

Reply:
Good point. There's no real reason (just an artifact of extraction a preliminary lemma).
Fixed now (to use mostly the first version).